### ISSUE TITLE
BigQuery Table Data Types Do Not Reflect Those of Postgres Source

### ISSUE THREAD

**Issue by SIGINT** [memiiso/debezium-server-bigquery#10]
I'm not sure if this is necessarily a limitation with Debezium or this specific Debezium BigQuery connector. I successfully built and deployed a Docker image running debezium-server-bigquery. I was able to replicate all of our Postgres relational data into BigQuery tables in near real-time but I noticed that a large chunk of the table's columns' data types in BigQuery don't line up with those in our Postgres schemas. I found [this documentation](documentation|https://debezium.io/documentation/reference/stable/connectors/postgresql.html#postgresql-basic-types) which explains the discrepancies I'm seeing. Basically all of the columns that are `TIMESTAMP` in Postgres are now `STRING` in BigQuery tables.

Is there any support built into the connector and/or Debezium for mapping the schema data types accordingly? What approaches have other folks used in this situation? It looks like it might be possible to write a custom table schema converter.

**Comment by ismailsimsek** [comment]
Good point seems like [debezium extracts it as string](https://debezium.io/documentation/reference/stable/connectors/postgresql.html#postgresql-basic-types) but it also has a special Semantic type. it should be possible to handle it properly.

- [debezium PG data type mapping list](https://debezium.io/documentation/reference/stable/connectors/postgresql.html#postgresql-basic-types)
- debezium to Bigquery [Consumer type mapping list](https://github.com/memiiso/debezium-server-bigquery/blob/master/debezium-server-bigquery-sinks/src/main/java/io/debezium/server/bigquery/DebeziumBigqueryEvent.java#L82)
